test/colony-funding.js

```diff
@@ -403,4 +407,367 @@ contract("Colony Funding", () => {
+
+      await colony.moveRemainingRewardPayoutFunds(payoutId);
+
+      const rewardPotBalanceAfterFinish = await colony.getPotBalance(0, otherToken.address);
+
+      assert.equal(rewardPotBalanceBeforeStart.sub(reward).toString(), rewardPotBalanceAfterFinish.toString());
+    });
+
+    it("should successfuly calculate reward with extreme differences between tokens and reputation", async () => {
+      const colonyName = getRandomString(7);
+      const tokenArgs = getTokenArgs();
+      const newToken = await Token.new(...tokenArgs);
+      await colonyNetwork.createColony(colonyName, newToken.address);
+      const address = await colonyNetwork.getColony.call(colonyName);
+
+      await newToken.mint(initialFunding.toString());
+      await newToken.transfer(userAddress1, initialFunding.toString());
+
+      await newToken.setOwner(address);
+      const newColony = await IColony.at(address);
+
+      await fundColonyWithTokens(newColony, otherToken, initialFunding.toString());
+      await newColony.mintTokens(initialFunding.toString());
+      await newColony.bootstrapColony([userAddress1, userAddress2, userAddress3], [30, 10, 20]);
+
+      const payoutId = await newColony.startNextRewardPayout.call(otherToken.address);
+      await newColony.startNextRewardPayout(otherToken.address);
+
+      const rewardPayoutInfo = await newColony.getRewardPayoutInfo(payoutId);
+      const amount = rewardPayoutInfo[1];
+
+      const totalSupply = await newToken.totalSupply();
+      const colonyTokens = await newToken.balanceOf(newColony.address);
+      const totalNewTokens = totalSupply.sub(colonyTokens);
+      const userTokens = await newToken.balanceOf(userAddress1);
+
+      const sharePercentage = userTokens
+        .mul(30)
+        .div(totalNewTokens * 60)
+        .sqrt();
+      const reward = amount.mul(sharePercentage).round();
+
+      await newColony.claimRewardPayout(payoutId, 30, 60, {
+        from: userAddress1
+      });
+
+      const rewardPayoutInfoAfterPayout = await newColony.getRewardPayoutInfo(payoutId);
+      const remaining = rewardPayoutInfoAfterPayout[2];
+
+      assert.equal(amount.sub(remaining).toString(), reward.round().toString());
+    });
+
+    const reputations = [
+      {
+        totalReputation: toBN(Math.floor(Math.random() * 1e3) + 1e2),
+        userReputation: toBN(Math.floor(Math.random() * 1e2)),
+        delta: 1
+      },
+      {
+        totalReputation: toBN(Math.floor(Math.random() * 1e6) + 1e4),
+        userReputation: toBN(Math.floor(Math.random() * 1e4)),
+        delta: 10
+      },
+      {
+        totalReputation: toBN(Math.floor(Math.random() * 1e12) + 1e8),
+        userReputation: toBN(Math.floor(Math.random() * 1e8)),
+        delta: 10
+      },
+      {
+        totalReputation: toBN(Math.floor(Math.random() * 1e21) + 1e16).mul(toBN(Math.floor(Math.random() * 1e3))),
+        userReputation: toBN(Math.floor(Math.random() * 1e16)),
+        delta: 100
+      }
+    ];
```

Comment:
I like these tests! Can we have more of them, for even more extreme values? 😃

Reply:
Sure, i just deleted them because i was getting `0` reward (As you mentioned above)